Java/core/src/main/java/com/amazon/randomcutforest/summarization/Center.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.summarization;
+
+import static java.lang.Math.exp;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import com.amazon.randomcutforest.util.WeightedIndex;
+
+/**
+ * the following class abstracts a single centroid representation of a group of
+ * points
+ */
+public class Center implements ICluster {
+
+    float[] representative;
+    double weight;
+    ArrayList<WeightedIndex<Integer>> assignedPoints;
+    double sumOfRadius;
+
+    double previousWeight = 0;
+    double previousSumOFRadius = 0;
+
+    Center(float[] coordinate, float weight) {
+        // explicitly copied because array elements will change
+        this.representative = Arrays.copyOf(coordinate, coordinate.length);
+        this.weight = weight;
+        this.assignedPoints = new ArrayList<>();
+    }
+
+    public static Center initialize(float[] coordinate, float weight) {
+        return new Center(coordinate, weight);
+    }
+
+    // adds a point; only the index to keep space bounds lower
+    // note that the weight may not be the entire weight of a point in case of a
+    // "soft" assignment
+    public void addPoint(int index, float weight, double dist) {
```

Comment:
It's seems strange that this class is missing (what I assume are) point store indexes and actual point values. What's the purpose of the indexes?

Reply:
The index is a number that currently refers to the index of the point in the sampledPoint list. It is absolutely true :) that this same index can refer to a pointStore index and we would have  clustering of the pointstore (among many other things). AT the current moment, the (int) index serves as a space saver (in the spirit of CompactRCF) and if the same point is (soft) allocated to multiple clusters then there is no duplication. So the PointStore plug and play is the next step :)

Reply:
This interface method implying something about the implementation. Does it makes sense to include it in a generic cluster interface?

Reply:
Fixed -- the interface is partitioned to a generic one and a more index specific one -- both of these can have usage down the road.